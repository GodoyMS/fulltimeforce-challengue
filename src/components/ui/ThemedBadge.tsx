import { StyleSheet, Text, View } from "react-native";
import React from "react";
import { LinearGradient } from "expo-linear-gradient";
import { useThemeContext } from "@context/themeContext";
import { ThemedText } from "./ThemedText";
interface ThemedBadgeProps {
  title: string;
}
const ThemedBadge = (props: ThemedBadgeProps) => {
  const { colors } = useThemeContext();
  return (
    <View style={{}}>
      <LinearGradient
        style={{ padding: 10, borderRadius: 50 }}
        colors={[colors[0], colors[1]]}
      >
        <ThemedText type="h4">{props.title}</ThemedText>
      </LinearGradient>
    </View>
  );
};

export default ThemedBadge;

const styles = StyleSheet.create({});
