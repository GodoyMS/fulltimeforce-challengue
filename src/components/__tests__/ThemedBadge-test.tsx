import ThemedBadge from "@components/ui/ThemedBadge";
import React from "react";
import renderer from "react-test-renderer";

// Mock the theme context to provide consistent colors
jest.mock("@context/themeContext", () => ({
  useThemeContext: () => ({
    colors: ["#123456", "#654321"],
  }),
}));

it("renders ThemedBadge correctly with a title", () => {
  const tree = renderer.create(<ThemedBadge title="Hello World" />).toJSON();
  expect(tree).toMatchSnapshot();
});
