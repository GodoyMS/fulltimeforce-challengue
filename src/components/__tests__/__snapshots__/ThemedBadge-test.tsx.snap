// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme test correctly 1`] = `
<Text
  style={
    [
      {
        "fontSize": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "fontFamily": "DMMedium",
      },
      undefined,
      {
        "color": "#ffffff",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`renders ThemedBadge correctly with a title 1`] = `
<View
  style={{}}
>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        4279383126,
        4284826401,
      ]
    }
    style={
      {
        "borderRadius": 50,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        [
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 16,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontFamily": "DMMedium",
          },
          undefined,
          {
            "color": "#ffffff",
          },
          undefined,
        ]
      }
    >
      Hello World
    </Text>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
